oat-sa/extension-pcisample
# fix: textreader destroys without removing tag and safe selection

Related to https://oat-sa.atlassian.net/browse/TR-2962

 - Update the text reader PCI in order to fix the destroy lifecycle


**What's wrong**

 - In review mode, every interaction receives a state and updates accordingly but due to a lack of `setState` method in the PCI specification, the PCI destroys itself and recreates with the new state (see https://github.com/oat-sa/tao-item-runner-qti-fe/blob/master/src/qtiCommonRenderer/renderers/interactions/pci/ims.js#L100-L110) 
 - But PCIs are coming with markup, inserted in the container before the PCI initialize. The text reader PCI was removing the whole content of the container, so setting the state was failing
  - We agree to fix that PCI but we need to figure out a solution so the markup is added again when instantiating the PCI

**How to test:**
 - launch a test review from that attached TAEs, for example this [review launch](https://devkit-pg-b46e5319fe637ed.cicd.gcp-eu.taocloud.org/platform/message/launch/lti-resource-link?registration=taoToolRegistration&user_list=c3po&launch_url=http://test-tr2962pci.playground.kitchen.it.taocloud.org:42531/ltiTestReview/ReviewTool/launch1p3?delivery%3Dhttp%253A%252F%252Ftest-tr2962pci.playground.kitchen.it.taocloud.org%252Ftao.rdf%2523i61dff3ee3ca9e40df1f61480b2f8079&claims=%7B%0D%0A%20%20%20%20%22https://purl.imsglobal.org/spec/lti/claim/roles%22:%20%5B%0D%0A%20%20%20%20%20%20%20%20%22http://purl.imsglobal.org/vocab/lis/v2/membership%23Learner%22%0D%0A%20%20%20%20%5D,%0D%0A%20%20%20%20%22https://purl.imsglobal.org/spec/lti/claim/custom%22:%20%7B%0D%0A%20%20%20%20%20%20%20%20%22execution%22:%20%22http://test-tr2962pci.playground.kitchen.it.taocloud.org/tao.rdf%23i61dff9ff420268621aefe354103dce%22%0D%0A%20%20%20%20%7D%0D%0A%7D)
 - test the migration from the previous version `php index.php '\oat\tao\scripts\tools\Migrations' -c migrate -v 'oat\pciSamples\migrations\Version202201111538504106_pciSamples' -e pciSamples`. This should publish the version 1.1.1 of the IMS textreader PCI 

 **Envs**:
 - [composer.json](https://github.com/oat-sa/kitchen-playground/blob/test/tr2962pci/files/delivery/composer.json)
 - [kitchen env](http://test-tr2962pci.playground.kitchen.it.taocloud.org:4253)
 - [devkit](https://devkit-pg-b46e5319fe637ed.cicd.gcp-eu.taocloud.org/)
